includes/Core/Util/WP_Context_Switcher_Trait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Util\WP_Context_Switcher_Trait
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+/**
+ * Trait for temporarily switching WordPress context, e.g. from admin to frontend.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait WP_Context_Switcher_Trait {
+
+	/**
+	 * Switches to WordPress frontend context if necessary.
+	 *
+	 * Context is only switched if WordPress is not already in frontend context. Context should only ever be switched
+	 * temporarily. Call the returned closure as soon as possible after to restore the original context.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Closure that restores context and returns true if context was restored or false otherwise.
+	 */
+	protected static function with_frontend_context() {
+		global $current_screen;
+
+		if ( ! is_admin() ) {
+			return function() {
+				return false;
+			};
```

Comment:
Not really a big deal, but if we're going to avoid switching the context in this case, we should probably still restore the current/original screen when this function is called. Even if the main function didn't change it, you could say this function should still behave the same way.

Reply:
Hmm there would be nothing to restore. One of the main reasons for me to implement this logic to bail early in the first place is that we shouldn't load the extra WP admin files if it isn't needed.

Reply:
It would "restore" it back to whatever it was – nothing (i.e. `null`), or a `WP_Screen` with `id` of `front` if one was set.

I don't want to get into the weeds too much on this as it is definitely an edge case, but here's what I mean

```php
$current_screen = null; // A front end context

$restore_context =  self::with_frontend_context();

set_current_screen( 'dashboard' );

$restore_context(); // returns false

$current_screen instanceof WP_Screen; // true; a WP_Screen in admin context
```

I don't think this is relevant for our current use but just thinking about keeping the behavior consistent (not necessarily supporting something somewhere "doing it wrong").

Reply:
Oh okay, that's fair. With the purpose to "restore" that makes sense, I was probably still more thinking about it from the original PR approach view which was closer to WP core's "switching things approach" where it only considers what was switched from originally but not any other unrelated "switches" of the same global.